src/nunit.analyzers/DiagnosticSuppressors/NonNullableFieldIsUninitializedSuppressor.cs

```diff
@@ -0,0 +1,81 @@
+#if !NETSTANDARD1_6
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Analyzers.Constants;
+
+namespace NUnit.Analyzers.DiagnosticSuppressors
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class NonNullableFieldIsUninitializedSuppressor : DiagnosticSuppressor
+    {
+        internal static readonly SuppressionDescriptor NullableFieldInitializedInSetUp = new SuppressionDescriptor(
+            id: AnalyzerIdentifiers.NonNullableFieldIsUninitialized,
+            suppressedDiagnosticId: "CS8618",
+            justification: "Field is initialized in SetUp or OneTimeSetUp method");
+
+        public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions { get; } =
+            ImmutableArray.Create(NullableFieldInitializedInSetUp);
+
+        public override void ReportSuppressions(SuppressionAnalysisContext context)
+        {
+            foreach (var diagnostic in context.ReportedDiagnostics)
+            {
+                SyntaxNode? node = diagnostic.Location.SourceTree?.GetRoot(context.CancellationToken)
+                                                                  .FindNode(diagnostic.Location.SourceSpan);
+
+                if (node is null)
+                {
+                    continue;
+                }
+
+                string fieldName = node.ToString();
```

Comment:
Probably my lack of understanding of `DiagnosticSuppressor`, but can we do something more strongly typed than a string for the field? (same comment for the other suppressor)

Reply:
CS8618 is raised for nulable fields, so we know that the SyntaxNode is a field. We don't actually care about the type.
But in this case, I do need to detect `this.field` as a separate assignment case: Code updated.

Reply:
For the actual null reference suppressor, we get the error reported on the expression. Again we don't care about the type. I check if the same expression that raises the null reference error is also used in the `Assert` call.
It fails when mixing expressions like in:
```
A? a = GetA(create);
Assert.That(a?.B?.Text, Is.Not.Null);
B b = a.B;
Assert.That(b.Text.Length, Is.GreaterThan(0));
```
Even though `b.Text` was tested as `a?.B?.Text` neither string comparison nor symbol analysis would help. Only if we could analyze the flow of assignments. The example is contrived, a more appropriate example would be:
```
A? a = GetA(create);
string? t = a?.B?.Text;
Assert.That(t, Is.Not.Null);
Assert.That(t.Length, Is.GreaterThan(0));
```

I am a bit disappointed that the following also fails:
```
A? a = GetA(create);
Assert.That(a?.B, Is.Not.Null);
B b = a.B;
string? t = b.Text;
Assert.That(t, Is.Not.Null);
Assert.That(t.Length, Is.GreaterThan(0));
```

The assignment `B b = a.B` where `b` is not nullable is suppressed correctly, but then the compiler still complains about `b` being possible `null` in `b.Text`. It looks like suppressors can only suppress an error, but not alter the compilers idea of what could be `null`.
It works when doing:
```
A? a = GetA(create);
B? b = a?.B;
Assert.That(b, Is.Not.Null);
string? t = b.Text;
Assert.That(t, Is.Not.Null);
Assert.That(t.Length, Is.GreaterThan(0));
```

Reply:
Thanks for the explanation. It helped a lot on my understanding.